astroid/helpers.py

```diff
@@ -199,3 +199,44 @@ def class_instance_as_index(node):
     except exceptions.InferenceError:
         pass
     return None
+
+
+def object_len(node, context=None):
+    """Infer length of given node object
+
+    :param Union[nodes.ClassDef, nodes.Instance] node:
+    :param node: Node to infer length of
+
+    :raises AstroidTypeError: If an invalid node is given
+        for an invalid type is returned
+    :raises InferenceError: If the given node cannot be inferred
+        or if multiple nodes are inferred
+    :rtype int: Integer length of node
+    """
+    inferred_node = safe_infer(node, context=context)
+    if inferred_node is None or inferred_node is util.Uninferable:
+        raise exceptions.InferenceError(node=node)
+    if inferred_node.qname() in ('builtins.str', 'builtins.bytes'):
```

Comment:
I'd suggest to check against concrete types, e.g `Dict, List, Tuple` rather than verifying their inner nodes. This way it will be more clear what types are these.

Reply:
More qname?

Reply:
`qname()` could work, as long as it's clear what nodes are we handling here.

Reply:
I think maybe in the future I could transform builtin methods to return correct values instead of being empty. Do you think this is easily do-able?

Reply:
Sorry I don't understand what you mean :)